ONSdigital/dp-dataset-exporter-xlsx
# Update XLSX service with encryption

### What

- Added a Makefile to handle vault setup
- Determines with version is published/unpublished
- Decrypts any unpublished CSV files
- Encrypts XLSX file if it is not published
- Integration with vault server
- Updated Unit tests
- Updated README

### How to review

- Run full stack
- Ensure dataset-exporter-xlsx is on `feature/private-public`
- Ensure dataset-exporter is on `feature/determine-state`
- Ensure dataset-api is on `feature/generate-downloads-on-publish`
- Ensure download service is running
- Import an instance
- When instance is edition-confirmed, verify private links are generated and that the links are encrypted
- When instance is published verify public links are generated which are not encrypted

### Who can review

Anyone
